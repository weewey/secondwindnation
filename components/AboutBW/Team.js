import React from 'react';
import Typography from 'material-ui/Typography';
import { withStyles } from 'material-ui/styles';
import PropTypes from 'prop-types';
import Grid from 'material-ui/Grid';
import Hidden from 'material-ui/Hidden';
import Button from 'material-ui/Button';
import Avatar from 'material-ui/Avatar';


const styleSheet = theme => ({
  root: {
    width: '100%',
    height: '60vh',
    textAlign: 'center',
    // backgroundColor: 'green',
    justifyContent: 'center',
    display: 'flex',
    alignItems: 'center',
    marginTop: 30,
  },
  para: {
    maxWidth: 1200,
    padding: 20,
    margin: 'auto',
  },
  button: {
    border: '1px solid black',
    backgroundColor: 'white',
    borderRadius: '2em',
    marginTop: 50,
    marginBottom: 50,
  },
  title: {
    // fontWeight: 500,
    color: 'black',
  },
  profile: {
    margin: 'auto',
  },
});

const members = [
  {
    name: 'Alan',
    title: 'Editor',
    initials: 'A',
  },
  {
    name: 'Chris',
    title: 'Development Coach',
    initials: 'C'
  },
  {
    name: 'Eugene',
    title: 'Performance Coach',
    initials: 'E',
  },
  {
    name: 'Melissa',
    title: 'Marcoms Strategist',
    initials: 'M',
  },
  {
    name: 'Romaine',
    title: 'Editor',
    initials: 'R',
  },
  {
    name: 'Shen',
    title: 'Development Coach',
    initials: 'S',
  },
  {
    name: 'Yew Wee',
    title: 'Chief Technology Officer',
    initials: 'YW',
  },
  {
    name: 'Z',
    title: 'Volunteer Photographer',
    initials: 'Z',
  },
  {
    name: 'Jian Hao',
    title: 'Web Developer',
    initials: 'JH',
  },
];

const teamList = members.map((member, index) => {
  return (
    <Grid item xs={4} sm={4} md={4} key={index}>
      <Avatar style={{ margin: 'auto' }}>{member.initials}</Avatar>
      <Typography type="title" style={{ marginTop: 10}}>{member.name}</Typography>
      <Typography type="subheading">{member.title}</Typography>
    </Grid>
  );
});

function Team(props) {
  const classes = props.classes;
  return (
    <div className={classes.root}>
      <div className={classes.para}>
        <Grid container spacing={40} align="center" justify="center">
          <Grid item xs={12}>
            <Typography type="display3" className={classes.title}>TEAM</Typography>
          </Grid>
          {teamList}
          {/* <Button raised className={classes.button}>Join Us!</Button> */}
        </Grid>
      </div>
    </div>
  );
}


Team.propTypes = {
  classes: PropTypes.object.isRequired,
};

teamList.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styleSheet)(Team);
