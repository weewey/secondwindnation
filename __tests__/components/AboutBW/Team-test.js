import React from 'react';
import renderer from 'react-test-renderer';

import Team from '../../../components/AboutBW/Team';

describe('Team component', () => {
  it('should render', () => {
    const team = renderer.create(<Team />,);
    const tree = team.toJSON();
    expect(tree).toMatchSnapshot();
  },
);
})
