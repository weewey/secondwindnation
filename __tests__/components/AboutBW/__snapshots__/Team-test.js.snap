// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Team component should render 1`] = `
<div
  className="Team-root-1"
>
  <div
    className="Team-para-2"
  >
    <div
      className="MuiGrid-typeContainer-6 MuiGrid-spacing-xs-40-23 MuiGrid-align-xs-center-12 MuiGrid-justify-xs-center-16"
    >
      <div
        className="MuiGrid-typeItem-7 MuiGrid-grid-xs-12-36"
      >
        <h1
          className="MuiTypography-root-89 MuiTypography-display3-91 Team-title-4"
        >
          TEAM
        </h1>
      </div>
      <div
        className="MuiGrid-typeItem-7 MuiGrid-grid-xs-4-28 MuiGrid-grid-sm-4-41 MuiGrid-grid-md-4-54"
      >
        <div
          className="MuiAvatar-root-111 MuiAvatar-colorDefault-112"
          style={
            Object {
              "margin": "auto",
            }
          }
        >
          A
        </div>
        <h2
          className="MuiTypography-root-89 MuiTypography-title-95"
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          Alan
        </h2>
        <h3
          className="MuiTypography-root-89 MuiTypography-subheading-96"
        >
          Editor
        </h3>
      </div>
      <div
        className="MuiGrid-typeItem-7 MuiGrid-grid-xs-4-28 MuiGrid-grid-sm-4-41 MuiGrid-grid-md-4-54"
      >
        <div
          className="MuiAvatar-root-111 MuiAvatar-colorDefault-112"
          style={
            Object {
              "margin": "auto",
            }
          }
        >
          C
        </div>
        <h2
          className="MuiTypography-root-89 MuiTypography-title-95"
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          Chris
        </h2>
        <h3
          className="MuiTypography-root-89 MuiTypography-subheading-96"
        >
          Development Coach
        </h3>
      </div>
      <div
        className="MuiGrid-typeItem-7 MuiGrid-grid-xs-4-28 MuiGrid-grid-sm-4-41 MuiGrid-grid-md-4-54"
      >
        <div
          className="MuiAvatar-root-111 MuiAvatar-colorDefault-112"
          style={
            Object {
              "margin": "auto",
            }
          }
        >
          E
        </div>
        <h2
          className="MuiTypography-root-89 MuiTypography-title-95"
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          Eugene
        </h2>
        <h3
          className="MuiTypography-root-89 MuiTypography-subheading-96"
        >
          Performance Coach
        </h3>
      </div>
      <div
        className="MuiGrid-typeItem-7 MuiGrid-grid-xs-4-28 MuiGrid-grid-sm-4-41 MuiGrid-grid-md-4-54"
      >
        <div
          className="MuiAvatar-root-111 MuiAvatar-colorDefault-112"
          style={
            Object {
              "margin": "auto",
            }
          }
        >
          M
        </div>
        <h2
          className="MuiTypography-root-89 MuiTypography-title-95"
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          Melissa
        </h2>
        <h3
          className="MuiTypography-root-89 MuiTypography-subheading-96"
        >
          Marcoms Strategist
        </h3>
      </div>
      <div
        className="MuiGrid-typeItem-7 MuiGrid-grid-xs-4-28 MuiGrid-grid-sm-4-41 MuiGrid-grid-md-4-54"
      >
        <div
          className="MuiAvatar-root-111 MuiAvatar-colorDefault-112"
          style={
            Object {
              "margin": "auto",
            }
          }
        >
          R
        </div>
        <h2
          className="MuiTypography-root-89 MuiTypography-title-95"
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          Romaine
        </h2>
        <h3
          className="MuiTypography-root-89 MuiTypography-subheading-96"
        >
          Editor
        </h3>
      </div>
      <div
        className="MuiGrid-typeItem-7 MuiGrid-grid-xs-4-28 MuiGrid-grid-sm-4-41 MuiGrid-grid-md-4-54"
      >
        <div
          className="MuiAvatar-root-111 MuiAvatar-colorDefault-112"
          style={
            Object {
              "margin": "auto",
            }
          }
        >
          S
        </div>
        <h2
          className="MuiTypography-root-89 MuiTypography-title-95"
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          Shen
        </h2>
        <h3
          className="MuiTypography-root-89 MuiTypography-subheading-96"
        >
          Development Coach
        </h3>
      </div>
      <div
        className="MuiGrid-typeItem-7 MuiGrid-grid-xs-4-28 MuiGrid-grid-sm-4-41 MuiGrid-grid-md-4-54"
      >
        <div
          className="MuiAvatar-root-111 MuiAvatar-colorDefault-112"
          style={
            Object {
              "margin": "auto",
            }
          }
        >
          YW
        </div>
        <h2
          className="MuiTypography-root-89 MuiTypography-title-95"
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          Yew Wee
        </h2>
        <h3
          className="MuiTypography-root-89 MuiTypography-subheading-96"
        >
          Chief Technology Officer
        </h3>
      </div>
      <div
        className="MuiGrid-typeItem-7 MuiGrid-grid-xs-4-28 MuiGrid-grid-sm-4-41 MuiGrid-grid-md-4-54"
      >
        <div
          className="MuiAvatar-root-111 MuiAvatar-colorDefault-112"
          style={
            Object {
              "margin": "auto",
            }
          }
        >
          Z
        </div>
        <h2
          className="MuiTypography-root-89 MuiTypography-title-95"
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          Z
        </h2>
        <h3
          className="MuiTypography-root-89 MuiTypography-subheading-96"
        >
          Volunteer Photographer
        </h3>
      </div>
      <div
        className="MuiGrid-typeItem-7 MuiGrid-grid-xs-4-28 MuiGrid-grid-sm-4-41 MuiGrid-grid-md-4-54"
      >
        <div
          className="MuiAvatar-root-111 MuiAvatar-colorDefault-112"
          style={
            Object {
              "margin": "auto",
            }
          }
        >
          JH
        </div>
        <h2
          className="MuiTypography-root-89 MuiTypography-title-95"
          style={
            Object {
              "marginTop": 10,
            }
          }
        >
          Jian Hao
        </h2>
        <h3
          className="MuiTypography-root-89 MuiTypography-subheading-96"
        >
          Web Developer
        </h3>
      </div>
    </div>
  </div>
</div>
`;
